processing/src/test/java/org/apache/druid/segment/UnnestStorageAdapterTest.java

```diff
@@ -300,7 +301,56 @@ public void test_pushdown_or_filters_unnested_and_original_dimension_with_unnest
       return null;
     });
   }
+  @Test
+  public void test_nested_filters_unnested_and_original_dimension_with_unnest_adapters()
+  {
+    final UnnestStorageAdapter unnestStorageAdapter = new UnnestStorageAdapter(
+        new TestStorageAdapter(INCREMENTAL_INDEX),
+        new ExpressionVirtualColumn(OUTPUT_COLUMN_NAME, "\"" + COLUMNNAME + "\"", null, ExprMacroTable.nil()),
+        null
+    );
+
+    final VirtualColumn vc = unnestStorageAdapter.getUnnestColumn();
+
+    final String inputColumn = unnestStorageAdapter.getUnnestInputIfDirectAccess(vc);
+
+    final OrFilter baseFilter = new OrFilter(ImmutableList.of(
+        new SelectorDimFilter(OUTPUT_COLUMN_NAME, "1", null).toFilter(),
+        new AndFilter(ImmutableList.of(
+            new SelectorDimFilter(inputColumn, "2", null).toFilter(),
+            new SelectorDimFilter(OUTPUT_COLUMN_NAME, "10", null).toFilter()
+        ))
+    ));
+
+    final OrFilter expectedPushDownFilter = new OrFilter(ImmutableList.of(
```

Comment:
Can we add a similar test case on AND of OR filters as well?

Reply:
yes

Reply:
adding more test case